src/npr/rules/2_rename.smk

```diff
@@ -70,30 +96,30 @@ rule rename_files:
                 passlist = glob.glob(
                     os.path.join('fastq','pass','*fastq.gz')
                 )
-                cmd = ['cat'] + passlist
-                with open(pass_out, 'w') as f:
-                    sp.call(cmd, stdout=f)
+                cat_list(passlist, pass_out, msg="No fail/*fastq.gz")
                 #for f in passlist:
                 #    os.remove(f)
                 #fail
                 logfile.write("failing fastq files.\n")
                 faillist = glob.glob('fastq/fail/*fastq.gz')
-                cmd = ['cat'] + faillist
-                with open(fail_out, 'w') as f:
-                    sp.call(cmd, stdout=f)
+                cat_list(faillist, fail_out, msg="No fail/*fastq.gz")
                 #for f in faillist:
                 #    os.remove(f)
+
                 logfile.write("Copy sequencing_summary to sample folder.\n")
-                shutil.copy(
-                    'fastq/sequencing_summary.txt',
-                    os.path.join(
-                        sampleid_dir, 
-                        'sequencing_summary.txt'
-                ))
+                ss_file = 'fastq/sequencing_summary.txt'
+                ss_new = os.path.join(sampleid_dir, 'sequencing_summary.txt')
+                if Path(ss_file).exists():
+                    shutil.copy( ss_file, ss_new )
+                    logfile.write('copied {} to {}'.format(ss_file, ss_new))
+                else:
+                    #Path(ss_new).touch()
+                    logfile.write('no such file {}. Check basecalling.'.format(ss_file))
```

Comment:
Should this be dealt with without letting the pipeline fail ?

Reply:
Indeed. I think it's better to fail here directly, rather than in the next step.
sys.exit(1) is now in place